homeassistant/components/incomfort/config_flow.py

```diff
@@ -102,6 +107,33 @@ def async_get_options_flow(
         """Get the options flow for this handler."""
         return InComfortOptionsFlowHandler()
 
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Prepare configuration for a DHCP discovered Intergas Gateway device."""
+        self._discovered_host = discovery_info.ip
+        self._async_abort_entries_match({CONF_HOST: self._discovered_host})
+        unique_id = format_mac(discovery_info.macaddress)
+        await self.async_set_unique_id(unique_id)
```

Comment:
Your unique id is the serial number right? If that's the case, this is wrong

Reply:
The gateway has no exposed serial number, it can expose multiple serial numbers for each boiler connected to the gateway.